src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+package com.geeksforless.scenario;
+
+import com.geeksforless.client.client.WorkerClient;
+import com.geeksforless.client.model.dto.scenario.*;
+import com.geeksforless.client.model.entity.CustomUserDetails;
+import com.geeksforless.client.service.scenario.impl.ScenarioServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.Authentication;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class ScenarioServiceImplTest {
+
+    @Mock
+    private WorkerClient workerClient;
+
+    @Mock
+    private Authentication authentication;
+
+    @InjectMocks
+    private ScenarioServiceImpl scenarioService;
+
+    @BeforeEach
+    void setUp() {
+        CustomUserDetails userDetails = new CustomUserDetails();
+        userDetails.setId(1L);
+        userDetails.setEmail("user@example.com");
+        userDetails.setPassword("password");
+        lenient().when(authentication.getPrincipal()).thenReturn(userDetails);
+    }
+
+    @Test
+    void testSendScenario() {
+        ScenarioFormDTO formData = new ScenarioFormDTO("[]", false);
+        ScenarioRequestDTO expectedRequest = ScenarioRequestDTO.builder()
+                .userId(1L)
+                .scenarioList(Collections.emptyList())
+                .proxyRequired(false)
+                .build();
+        scenarioService.sendScenario(formData, authentication);
```

Comment:
dont you need to add *doNothing().when(client.postScenarios(any(SendScenarioRequest.class)))*??

Reply:
The doNothing().when() function is used to specify the behavior of mocks when you need to configure them to do something (for example, throw an exception, return certain values, etc.) when a method is called. But in our case, we only check that the method was called with the expected arguments using verify(workerClient).postScenarios(expectedRequest).